src/WordPress/Hooks.php

```diff
@@ -326,20 +334,20 @@ public function getPostRelatedLinks($postId)
             }
             $listofurls = array_merge(
                 $listofurls,
-                array_unique(array_filter($attachmentUrls))
+                $attachmentUrls
             );
         }
 
+        // Clean array and get unique values
+        $listofurls = array_values(array_filter(array_unique($listofurls)));
+
         // Purge https and http URLs
         if (function_exists('force_ssl_admin') && force_ssl_admin()) {
```

Comment:
I was looking over this doubling of urls based on ssl.   Seems to me it would make sense to add this bit of code to the getPostRelatedLinks.  Any reason to add all the http links only to possibly remove them later?

```
 $hasAlwaysUseHTTPSOverrideDisabled = $this->pageRuleContains($activePageRules, "always_use_https", "off");
  if ($this->zoneSettingAlwaysUseHTTPSEnabled($zoneTag) && !$hasAlwaysUseHTTPSOverrideDisabled) {
      $this->logger->debug("always_use_https is enabled without page rule overrides present, removing HTTP based URLs");
      $urls = array_filter($urls, array($this, "urlIsHTTPS"));
  }
```

Reply:
this is to ensure that even after filters are applied, we are cleaning up the HTTP only URLs that will never get cached.